### PR TITLE
ci: Disable checks unrunable @ pre-commit.ci

### PR DESCRIPTION
**Pull request by webknjaz**
Put an `x` into the box if that apply:

- [ ] This PR introduces breaking change.
- [ ] This PR fixes a bug.
- [ ] This PR adds new functionality.
- [ ] This PR enhances existing functionality.
- [x] This is a contributor/maintainer facing change that in no way affects the end-users

### Description of your changes

Enabling faster feedback loop for new contributors by enabling https://pre-commit.ci in this repo. 

The only visible effect of this is when said app is installed and the service is running the checks, it would skip the checks that rely on the external executables that might be missing in runtime. It's necessary since that testing environment is controlled by someone else.

### How can we test changes

@antonbabenko  go to https://github.com/apps/pre-commit-ci/installations/select_target, install the app into this repo and then merge the PR.

### CODE REVIEW

**Comment by MaxymVlasov** [comment]
We have pre-commit via GHA, and it runs all hooks specified in .pre-commit-config, not just some of them

**Comment by webknjaz** [comment]
Alternatively, there's https://github.com/apps/pre-commit-ci-lite/installations/new + https://pre-commit.ci/lite.html that implements backpushes.

**Comment by yermulnik** [comment]
M-m-m, I definitely would suggest to add more context for other contributors, including a brief explanation added as a comment to the code that is changing for future reference and for posterity.

**Comment by webknjaz** [comment]
Done.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,3 +1,12 @@
+---
+
```
Comment (MaxymVlasov) [inline .pre-commit-config.yaml:2]: It's a single YAML document, so no need to add this boilerplate 
```suggestion
```

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -59,3 +68,5 @@ repos:
   - id: prettier
     # https://prettier.io/docs/en/options.html#parser
     files: '.json5$'
+
+...
```
Comment (MaxymVlasov) [inline .pre-commit-config.yaml:72]: Same for [end file notation](https://stackoverflow.com/a/66542407). TBH, that's first time when I see that syntax, and I saw much more YAMLs than I'd like to have.

Can't imagine situation when someone will try to merge this file with anything else and don't implement basic separation between files

```suggestion
```

File: .pre-commit-config.yaml (hunk 1 above)
Comment (yermulnik) [reply .pre-commit-config.yaml:2]: That also designates a start of YAML document =)
Though I agree: no need for it if it is not required.

File: .pre-commit-config.yaml (hunk 2 above)
Comment (yermulnik) [reply .pre-commit-config.yaml:72]: +1

**Comment by webknjaz** [comment]
pre-commit.ci run

**Comment by webknjaz** [comment]
It's not marked as required in branch protection, though.

**Comment by antonbabenko** [comment]
@webknjaz Updated.
image

**Comment by webknjaz** [comment]
This check is now failing in other PRs and will keep failing until this PR is merged.

**Comment by antonbabenko** [comment]
This PR is included in version 1.97.0 :tada:

**Pull request merged**